EventFilter/SiPixelRawToDigi/plugins/SiPixelDigiToRaw.cc

```diff
@@ -130,6 +134,25 @@ void SiPixelDigiToRaw::produce( edm::StreamID, edm::Event& ev,
 
   LogDebug("SiPixelDigiToRaw") << cache->cablingTree_->version();
 
+  PixelDataFormatter::BadChannels badChannels;
+  edm::Handle<PixelFEDChannelCollection> pixelFEDChannelCollectionHandle;
+  if (ev.getByToken(theBadPixelFEDChannelsToken, pixelFEDChannelCollectionHandle)){
+    for (auto const& fedChannels: *pixelFEDChannelCollectionHandle) {
+      PixelDataFormatter::DetBadChannels detBadChannels;
+      for(const auto& fedChannel: fedChannels) {
+	sipixelobjects::CablingPathToDetUnit path={fedChannel.fed, fedChannel.link, 1};
+	if (cache->cablingTree_->findItem(path)!=nullptr) {
+	  detBadChannels.push_back(fedChannel);
+	} else {
+	  edm::LogError("SiPixelDigiToRaw")
```

Comment:
Basically should never happen if the payload based on which the digitizer kills modules lists only existing channels. This is just a safety net here.

Reply:
How about throwing an exception then?

Reply:
In regular production, we would definitely want to notice if this happens (frequent warning would probably trigger attention), but not sure if we would want to enforce only one possible usage of the code by preventing any exotic used cases.
E.g. should one be able to run DigiToRaw+RawToDigi in private jobs for single FEDs while processing a file with a digi collection produced once for the full detector?

Reply:
> E.g. should one be able to run DigiToRaw+RawToDigi in private jobs for single FEDs while processing a file with a digi collection produced once for the full detector?

I don't know. My experience is just that log messages are usually ignored until their amount becomes excessive. One option would be to have a boolean flag to control whether to throw or not (default being to throw).

Reply:
Hi @makortel IMHO, throwing here would be a bit of an overkill, the check that Viktor does here, if failed, doesn't lead to corrupt data or to random failures. It just checks that whatever we have put in the payload for the simulation *does* correspond to an actual FED channel, i.e. logs an error if we are trying to kill anything which is not physically a FED channel (e.g. single-ROCs, etc.), so it's more a question of book-keeping of inefficiency source. Also there are other failsafes in the payload producers that are meant to avoid fall in this case and indeed it should never happen.